core/src/main/scala/sttp/tapir/server/interpreter/EncodeOutputs.scala

```diff
@@ -37,8 +38,9 @@ class EncodeOutputs[B, S](rawToResponseBody: ToResponseBody[B, S]) {
       case EndpointOutput.FixedStatusCode(sc, _, _) => ov.withStatusCode(sc)
       case EndpointIO.FixedHeader(header, _, _)     => ov.withHeader(header.name, header.value)
       case EndpointIO.Body(rawBodyType, codec, _) =>
+        val maybeCharset = if (codec.format.mediaType.mainType.equalsIgnoreCase("text")) charset(rawBodyType) else None
         ov.withBody(headers => rawToResponseBody.fromRawValue(encoded[Any], headers, codec.format, rawBodyType))
-          .withDefaultContentType(codec.format, charset(codec.format.mediaType, rawBodyType))
+          .withDefaultContentType(codec.format, maybeCharset)
```

Comment:
here the question is the same - should we check if whatever is in the `Accept` of the request matches what we want to return? Or just return regardless? Either way, behavior should be consistent with that of `oneOf`